source/nodejs/adaptivecards/src/carousel.ts

```diff
@@ -419,6 +452,27 @@ export class Carousel extends Container {
     }
 }
 
+export class CarouselEvent {
+    type : Enums.CarouselInteractionEvent;
+    carouselId : string | undefined;
+    activeCarouselPageId : string | undefined;
+    activeCarouselPageIndex : number | undefined;
+    constructor(carousel : Carousel) {
+        this.carouselId = carousel.id;
+        this.activeCarouselPageIndex = carousel.currentPageIndex;
+        this.activeCarouselPageId = (this.activeCarouselPageIndex != undefined) ? carousel.getItemAt(this.activeCarouselPageIndex).id : undefined; 
+        this.type = carousel.currentEventType;
+    }
+}
+
+function raiseCarouselEvent(carousel: Carousel, eventType : Enums.CarouselInteractionEvent) {
+    const onCarouselEventHandler = AdaptiveCard.onCarouselEvent;
+    if (onCarouselEventHandler && eventType == Enums.CarouselInteractionEvent.Pagination) {
+        onCarouselEventHandler(new CarouselEvent(carousel));
+    }
+    carousel.currentEventType = eventType;
```

Comment:
I saw you use currentEventType to constructor CarouselEvent. will `onCarouselEventHandler(new CarouselEvent(..., eventType))` work?

Reply:
i tested it, it works.